piet/src/render_context.rs

```diff
@@ -170,3 +180,17 @@ pub trait RenderContext {
     /// aspect ratios don't match.
     fn draw_image(&mut self, image: &Self::Image, rect: impl Into<Rect>, interp: InterpolationMode);
 }
+
+/// A brush that can be dependent on the geometry being drawn.
+pub trait IBrush<P: RenderContext>
+where
+    P: ?Sized,
+{
+    fn make_brush<'a>(&'a self, piet: &mut P, bbox: impl FnOnce() -> Rect) -> Cow<'a, P::Brush>;
```

Comment:
Would it be possible for this function to take ownership of self, and return an owned brush? I'm not sure if this makes sense, but it might make for a cleaner signature, and possibly get rid of the `Cow` business.

Reply:
Thinking a bit more I guess not; this would prevent reusing cooked brushes.

Reply:
A goal was for it to take an existing `&Brush`. Taking ownership of self seems problematic to me, as a lot of the time you're going to get these resources from somewhere (env, internal widget state).

This trait has a complex type, for sure, but it's not something users are going to have to deal with much; I think of it as similar to the hashmap types.